### ISSUE TITLE
Review MeshGateway and Gateways in general

### ISSUE THREAD

**Issue by lahabana** [kumahq/kuma#6741]
### Description

https://github.com/kumahq/kuma/blob/ae9330cceffe6b0747f8d7d5b7935a33504c3eab/pkg/core/resources/apis/mesh/zz_generated.resources.go#L973

This means it's in /policies which at the moment is only used for the GUI.

Feels like maybe this isn't a policy just like `ExternalService` isn't one. Should we change this now? Should we wait until we define a `Gateway` dataplane?

Another question is should there be a `provided` option in the MeshGateway?
In a world where gateway instances don't create `Dataplane` but their own resource (like ZoneIngress or ZoneEgress) what do we do?
What's the transition like?

**Comment by lahabana** [comment]
Once the GUI view is done we can remove MeshGateway from the /policies endpoint as it's not a policy

**Comment by michaelbeaumont** [comment]
https://github.com/kumahq/kuma/issues/9268

**Comment by lahabana** [comment]
Right duplicate of #9268 then :)
